app/src/main/java/org/mozilla/fenix/perf/ProfilerMarkers.kt

```diff
@@ -19,4 +20,18 @@ object ProfilerMarkers {
             profiler?.addMarker("onPreDraw", "expected first frame via HomeActivity.onStart")
         }
     }
+
+    fun addForDispatchTouchEvent(profiler: Profiler?, ev: MotionEvent?) {
+        // We only run this if the profiler is active to minimize any possible delay on touch events.
+        if (profiler?.isProfilerActive() == true) {
```

Comment:
Are there places when we want to call `addMarker` without this check first? Could we put this inside the component?

Reply:
Under the hood, `addMarker` checks the equivalent of `isProfilerActive` before adding the marker: https://searchfox.org/mozilla-central/rev/e98b9b6a20123cdb1d58ca683de123ce61f1e1a6/mobile/android/geckoview/src/main/java/org/mozilla/gecko/GeckoJavaSampler.java#354-357

`isProfileActive` is best used to avoid doing expensive operations needed to add the marker if the profiler isn't actually on. In this case, I guess I'm just avoiding the `when` statement, which probably isn't expensive so perhaps this wasn't worth it. 😅 I was originally concerned with how many `ACTION_MOVE` events are triggered.

Reply:
@jonalmeida What do you mean put it inside the component? Which component?

Reply:
> @jonalmeida What do you mean put it inside the component? Which component?

Sorry, I meant in the browser-gecko component where the [profiler is exposed](https://github.com/mozilla-mobile/android-components/blob/main/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/profiler/Profiler.kt) through, but your comment above shows that we no-op out early inside the runtime  already, and your logic to avoid extra operations is reasonable too. 🙂